provider/awssd/aws_sd.go

```diff
@@ -79,17 +79,26 @@ type AWSSDProvider struct {
 	cleanEmptyService bool
 	// filter services for removal
 	ownerID string
+	// tags to be added to the service
+	tags []sdtypes.Tag
 }
 
 // NewAWSSDProvider initializes a new AWS Cloud Map based Provider.
-func NewAWSSDProvider(domainFilter endpoint.DomainFilter, namespaceType string, dryRun, cleanEmptyService bool, ownerID string, client AWSSDClient) (*AWSSDProvider, error) {
+func NewAWSSDProvider(domainFilter endpoint.DomainFilter, namespaceType string, dryRun, cleanEmptyService bool, ownerID string, tags map[string]string, client AWSSDClient) (*AWSSDProvider, error) {
 	p := &AWSSDProvider{
 		client:              client,
 		dryRun:              dryRun,
 		namespaceFilter:     domainFilter,
 		namespaceTypeFilter: newSdNamespaceFilter(namespaceType),
 		cleanEmptyService:   cleanEmptyService,
 		ownerID:             ownerID,
+		tags: func() []sdtypes.Tag {
+			awsTags := make([]sdtypes.Tag, 0, len(tags))
+			for k, v := range tags {
+				awsTags = append(awsTags, sdtypes.Tag{Key: aws.String(k), Value: aws.String(v)})
+			}
+			return awsTags
+		}(),
```

Comment:
This PR is lacking some test on NewAWSSDProvider (see newTestAWSSDProvider). Otherwise, it lgtm.

Reply:
I have moved this to a dedicated function and added a test.

Reply:
thanks :+1: